src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -49,7 +49,13 @@ public class PatchMapDrawer : Control
         get => map;
         set
         {
-            map = value ?? throw new ArgumentNullException(nameof(value), "setting to null not allowed");
+            if (value == null)
+                throw new ArgumentNullException(nameof(value), "setting to null not allowed");
+
+            if (map == value)
```

Comment:
This seems like good idea, though, would you or someone else be able to check why the patch extinction drawer repeatedly sets the map property? That would be a good place to also fix.

Reply:
```
PatchMapDrawer.set_Map() at D:\source\repos\Thrive\src\microbe_stage\editor\PatchMapDrawer.cs:line 53
PatchExtinctionBox.set_Map() at D:\source\repos\Thrive\src\microbe_stage\gui\PatchExtinctionBox.cs:line 27
StageHUDBase<MicrobeStage>.ShowPatchExtinctionBox()
StageBase<Microbe>.PlayerExtinctInPatch()
MicrobeStage.PlayerExtinctInPatch()
StageBase<Microbe>.HandlePlayerRespawn()
StageBase<Microbe>._Process()
MicrobeStage._Process()
```

Reply:
So the ShowPatchExtinctionBox should be modified to be called just once only if the extinction screen is not open?

Reply:
I think so, but as I stated, it causes a crash.

Reply:
Could someone help out here? If it takes more than a few days, I can take a look at this problem next week.

Reply:
I got it working.

Reply:
Seems like a good approach. I'll try to ask for testing on this, especially if multiple patch extinctions happen in a row to make sure there's no bugs.